### PR TITLE
[json-rpc] get_account by version

### PR DESCRIPTION
**Pull request by xli**
## Motivation

There are use cases require getting account data at same version, e.g. sum of child VASPs account balances.
Batch request may not fit for the case because the number of child VASPs maybe exceed the batch size limit.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

Integration test.

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Chaning JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

File: json-rpc/API-CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,11 @@ Please add the API change in the following format:
 
 ```
 
+## 2021-03-18 Add optional parameter `version` to `get_account` method
+
+Get account data by version.
+
+
```
Comment (davidiw) [inline json-rpc/API-CHANGELOG.md:20]: extra whitespace?

File: json-rpc/tests/integration_test.rs [hunk 2]
```diff
@@ -339,6 +339,45 @@ fn create_test_cases() -> Vec<Test> {
...
+                            "compliance_key": "",
+                            "compliance_key_rotation_events_key": format!("0000000000000000{}", address),
+                            "expiration_time": 18446744073709551615_u64,
+                            "human_name": "Novi 0",
+                            "num_children": 0,
+                            "type": "parent_vasp"
+                        },
+                        "sent_events_key": format!("0300000000000000{}", address),
+                        "sequence_number": 0
+                    }),
```
Comment (davidiw) [inline json-rpc/tests/integration_test.rs:377]: this test doesn't seem to be too useful as there's nothing here about version afaiu, don't we really just care that the version in the result is correctly set? also it seems like the json blob should be a bunch of predefined fields but things like expiration date, human name, and balances are not constants.

**Review by davidiw — COMMENTED** [review]
I think if I understand it correctly we didn't have a test for get_account? PR looks okay.

File: json-rpc/tests/integration_test.rs (hunk 2 above)
Comment (xli) [reply json-rpc/tests/integration_test.rs:377]: the test first find transaction version of account sequence number, then use the transaction version - 1 to find back the older version account. This ensures we can find back older version by version param.

the test is running with a single node with db included, it is hard to find out what is version the account is created or before last change.

the test above, tests same account and get back latest account info.
the difference between them are balances and sequence_number.

File: json-rpc/API-CHANGELOG.md (hunk 1 above)
Comment (xli) [reply json-rpc/API-CHANGELOG.md:20]: one extra line won't have difference after rendered. sometime I just gave 2 lines at the end in MD for better reading.

Comment (xli) [reply json-rpc/API-CHANGELOG.md:20]: extra line in md file is fine when rendered. I added one extra line at the end of section for easier reading.

File: json-rpc/tests/integration_test.rs (hunk 2 above)
Comment (xli) [reply json-rpc/tests/integration_test.rs:377]: the test first finds the first transaction version of the account (account sequence number 0),
then use the (version - 1) as transaction version to look up account, which gives the account data when it is set up.
Mostly to ensure we can look up previous version.
I'm not sure what you mean by "be a bunch of predefined fields but things like expiration date, human name, and balances are not constants.", these data all set up by test env.

the response version is in header or response body, every response contains it. but it maybe nice to have in Account response body too, we can add.

**Comment by xli** [comment]
> I think if I understand it correctly we didn't have a test for get_account? PR looks okay.

The test above my new test is a 'get_account' test, error cases are covered by unit tests.

File: json-rpc/src/methods.rs [hunk 3]
```diff
@@ -302,11 +304,7 @@ async fn get_account(
 /// Can be used to verify that target Full Node is up-to-date
 async fn get_metadata(service: JsonRpcService, request: JsonRpcRequest) -> Result<MetadataView> {
     let chain_id = service.chain_id().id();
-    let version = if !request.params.is_empty() {
-        request.parse_version_param(0, "version")?
-    } else {
-        request.version()
-    };
+    let version = request.parse_version_param(0, "version")?;
```
Comment (bmwill) [inline json-rpc/src/methods.rs:307]: I'm assuming that this logic happens in this function (handling the case where its not provided?)

File: json-rpc/types/src/views.rs [hunk 4]
```diff
@@ -140,6 +140,7 @@ pub struct AccountView {
     pub delegated_withdrawal_capability: bool,
     pub is_frozen: bool,
     pub role: AccountRoleView,
+    pub version: Option<u64>,
```
Comment (bmwill) [inline json-rpc/types/src/views.rs:143]: Do you want to add an annotation to this to not render it if its None?

File: json-rpc/src/methods.rs (hunk 3 above)
Comment (xli) [reply json-rpc/src/methods.rs:307]: yes, it handles the case given parameter index is > len(request params)

File: json-rpc/types/src/views.rs (hunk 4 above)
Comment (xli) [reply json-rpc/types/src/views.rs:145]: Added. It's nice to have. As server side constructor is not optional, and will always render.

**Review by bmwill — APPROVED** [review]

**Comment by kchalkias** [comment]
Thanks @xli, this is very useful for the proof-of-assets story. When your PR is merged, we'll update https://github.com/diem/diem/blob/main/client/assets-proof/src/lib.rs#L586 by invoking your api instead of `get_account_state_with_proof`.

**Pull request merged**